qiskit/pulse/instructions/call.py

```diff
@@ -38,35 +38,87 @@ def __init__(self, subroutine,
         .. note:: Inline subroutine is mutable. This requires special care for modification.
 
         Args:
-            subroutine (Schedule): A program subroutine to be referred to.
+            subroutine (Union[Schedule, ScheduleBlock]): A program subroutine to be referred to.
             value_dict: Mapping of parameter object to assigned value.
             name: Unique ID of this subroutine. If not provided, this is generated based on
                 the subroutine name.
-        """
-        if name is None:
-            name = f"{self.prefix}_{subroutine.name}"
 
-        super().__init__((subroutine,), None,
-                         channels=tuple(subroutine.channels),
-                         name=name)
-        if value_dict:
-            self.assign_parameters(value_dict)
+        Raises:
+            PulseError: If subroutine is not valid data format.
+        """
+        from qiskit.pulse.schedule import ScheduleBlock, Schedule
+
+        if not isinstance(subroutine, (ScheduleBlock, Schedule)):
+            raise PulseError(f'Subroutine {subroutine} is not acceptable data format.')
+
+        # initialize parameter template
+        # TODO remove self._parameter_table
+        self._arguments = dict()
+
+        if subroutine.is_parameterized():
+            for param_expr in subroutine.parameters:
```

Comment:
I'm not totally understanding why `Call` is the only place that has parameter-related logic that can't be handled by the ParameterManager

Reply:
Because `Call` separately store the parameters and program. This is to reuse program, i.e. subroutine, for different parameter sets. For example,
```python
with build() as sched:
   call(my_subroutine, amp=0.1)
   call(my_subroutine, amp=0.3)
```
This can be translated into program
```
{
   'library': {'subroutine1': Schedule((0, pulse.Play(Gaussian(160, amp, 40), DriveChannel(0)))}, 
   'program': [('subroutine, {amp: 0.1}), ('subroutine, {amp: 0.3})]
}
```
which will be more efficient than having
```
{
   'program': Schedule((0, pulse.Play(Gaussian(160, 0.1, 40), DriveChannel(0)), (0, pulse.Play(Gaussian(160, 0.1, 40), DriveChannel(0))))
}
```

To realize this behavior, it internally generates parameter table and decouple assigned parameters from subroutine. Parameter assignment just updates this parameter table, i.e. `Call._arguments`.